spec/requests/rack_attack_spec.rb

```diff
@@ -25,6 +25,30 @@ def unique_user_params
     expect(response).to have_http_status(:redirect)
   end
 
+  it "successful response does not include retry-after header" do
+    get root_path, env: { "REMOTE_ADDR" => Faker::Internet.unique.ip_v4_address }
+    expect(response).to have_http_status(:ok)
+    expect(response.headers["Retry-After"]).to be_nil
+  end
+
+  context "when there have been max number of requests from an IP address" do
+    let(:ip) { Faker::Internet.unique.ip_v4_address }
+
+    before do
+      ArchiveConfig.RATE_LIMIT_NUMBER.times do
+        get root_path, env: { "REMOTE_ADDR" => ip }
+      end
+    end
+
+    it "response to the next attempt from the same IP includes retry-after header" do
+      get root_path, env: { "REMOTE_ADDR" => ip }
+      expect(response).to have_http_status(:too_many_requests)
+      expect(response.headers["Retry-After"]).not_to be_nil
```

Comment:
You can omit this check since `nil..to_i` is 0 and covered by the check after.

Also, since time is frozen in these tests, is this header value equal exactly e.g. `ArchiveConfig.RATE_LIMIT_PERIOD`? If so, we can check for that.

Reply:
Checking for exact equality resulted in failed tests, so I guess they are not (always?) equal. I'm not familiar enough with the code to know why that is.

Reply:
I don't know either, but checking for a range is fine.